tests/socket_zep/main.c

```diff
@@ -116,7 +104,6 @@ int main(void)
     _main_pid = thread_getpid();
 
     test_init();
-    test_send__iolist_NULL();
```

Comment:
Why did you remove this part of the test ?

Reply:
With the radio HAL sending a `NULL` buffer will no longer result in any data being transmitted. Previously this would send a header-only packet (and this test would verify that).

Reply:
maybe:

    Elements of this list may have iolist_t::iol_size == 0 and (in this case only) iolist_t::iol_data == 0

should be tested instead

Reply:
I think this makes no sense. We are on layer 2 here, so for a real device this would mean no data on the bus / air.
Only with the (old) ZEP implementation this would write the ZEP header (that wraps the IEEE 802.15.4 frame).

Mind you that we are using the sub-mac now which does make assumptions about the contents being transmitted (namely them being a IEEE 802.15.4 frame) - we'd now need to add checks there, e.g in `sys/net/link_layer/ieee802154/submac.c:414` where it checks if the ACK_REQ bit is set - for a case that will never happen in reality.